lib/skate_web/channels/alerts_channel.ex

```diff
@@ -4,18 +4,16 @@ defmodule SkateWeb.AlertsChannel do
   alias Realtime.Server
   alias Util.Duration
 
-  @impl Phoenix.Channel
-  def join("alerts:route:" <> route_id, _message, socket) do
+  use SkateWeb.AuthenticatedChannel
+  @impl SkateWeb.AuthenticatedChannel
+  def join_authenticated("alerts:route:" <> route_id, _message, socket) do
     alerts = Duration.log_duration(Server, :subscribe_to_alerts, [route_id])
     {:ok, %{data: alerts}, socket}
   end
 
   @impl Phoenix.Channel
   def handle_info({:new_realtime_data, lookup_args}, socket) do
```

Comment:
**thought:** I wonder if we could similarly wrap `handle_info` and also get rid of some code duplication there...